nixos/modules/config/update-users-groups.pl

```diff
@@ -16,7 +16,7 @@
 
 sub updateFile {
     my ($path, $contents, $perms) = @_;
-    write_file("$path.tmp", { binmode => ':utf8', perms => $perms // 0644 }, $contents);
+    write_file("$path.tmp", { binmode => ':utf8', perms => $perms // 0644, atomic => 1 }, $contents);
     rename("$path.tmp", $path) or die;
```

Comment:
If we use this flag I don't see a point of having an additional rename step, which was also meant to make the write atomic-ish (no really so when we have multiple writers).

```suggestion
    write_file("$path", { binmode => ':utf8', perms => $perms // 0644, atomic => 1 }, $contents);
```

Reply:
The `or die` is correct in the error case it returns undef: https://github.com/perhunter/slurp/blob/d909fac92e6eb0592362c8337c7abc7f4693ac94/lib/File/Slurp.pm#L602